### PR TITLE
kebab casing and documentation for BQSR tools and a few others.

### PR DESCRIPTION
**Pull request by lucidtronix**
@sooheelee Kebab-cased and added to the docs.  I believe there are WDLs in production that this will break so we should proceed with caution...

### CODE REVIEW

**Comment by droazen** [comment]
@jamesemery Can you have a look at this one? I know you had some concerns about BQSR arg renaming, due to the fact that some arg names are stored in the recalibration tables.

**Comment by jamesemery** [comment]
I was looking into this myself as part of other argument renaming tasks. I don't see that you have changed any of the recalibration table arguments checked into the repo, namely in the test resources /tools/BQSR. (eg. `src/test/resources/org/broadinstitute/hellbender/tools/BQSR/HiSeq.1mb.1RG.lowMaxCycle.table`) You should probably take a look at this and other .table files and maybe update them to have more up to date summary statistics, as it could indicate that the gatk is writing its output with the old argument styles as these are used in integration tests.

**Comment by sooheelee** [comment]
Thanks for this work @lucidtronix! I'll be reviewing the doc portion, so you'll need to ask a developer to review the kebabbing. Looks like @jamesemery has been conscripted.

**Comment by ldgauthier** [comment]
@lucidtronix the beautiful thing about using Docker is that production is pinned to a specific image, but once we're done with all the arg changes I'll fix the production WDLs and we'll run the agonizingly long production test suite. In the unlikely event we need a hotfix in the meantime, we'll just have to branch from the last commit before renaming.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java [hunk 1]
```diff
@@ -33,16 +33,68 @@
...
+ *   -o recal_data.table
+ * </pre>
+ */
+
 @CommandLineProgramProperties(
-        summary = "Base Quality Score Recalibration (BQSR) -- Generates recalibration table based on various user-specified covariates (such as read group, reported quality score, machine cycle, and nucleotide context).",
-        oneLineSummary = "BaseRecalibrator on Spark",
+        summary = BaseRecalibratorSpark.USAGE_SUMMARY,
+        oneLineSummary = BaseRecalibratorSpark.USAGE_ONE_LINE_SUMMARY,
         programGroup = SparkProgramGroup.class
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Program group should be ReadDataProgramGroup.class. Again, the new program groups are summarized in the `1217Changes` tab of the Google Spreadsheet: https://docs.google.com/spreadsheets/d/19SvP6DHyXewm8Cd47WsM3NUku_czP2rkh4L_6fd-Nac/edit?usp=sharing

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java [hunk 2]
```diff
@@ -33,16 +33,68 @@
...
 import java.util.List;
 
+/**
+ * Spark version of the first pass of the base quality score recalibration.
+ * Generates a recalibration table based on various covariates
+ * (such as read group, reported quality score, machine cycle, and nucleotide context).
+ *
+ * <p>
+ * This walker is designed to work as the first pass in a two-pass pipeline. It does a by-locus traversal operating
+ * only at sites that are not in dbSNP. We assume that all reference mismatches we see are therefore errors and indicative
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: dbSNP --> the known-sites resource. 

Tis a bit antiquated to keep referring to dbSNP when we now have resources such as gnomAD and ExAc.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java [hunk 3]
```diff
@@ -33,16 +33,68 @@
 import java.io.PrintStream;
 import java.util.List;
 
+/**
+ * Spark version of the first pass of the base quality score recalibration.
+ * Generates a recalibration table based on various covariates
+ * (such as read group, reported quality score, machine cycle, and nucleotide context).
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Generates a recalibration table based on various covariates (such as read group, reported quality score, machine cycle, and nucleotide context). 

-->

Generates a recalibration table based on various covariates. The default covariates are read group, reported quality score, machine cycle, and nucleotide context.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java [hunk 4]
```diff
@@ -33,16 +33,68 @@
...
+ *
+ * <p>
+ * This walker is designed to work as the first pass in a two-pass pipeline. It does a by-locus traversal operating
+ * only at sites that are not in dbSNP. We assume that all reference mismatches we see are therefore errors and indicative
+ * of poor base quality. This walker generates tables based on various user-specified covariates (such as read group,
+ * reported quality score, cycle, and context). Since there is a large amount of data one can then calculate an empirical
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: --> This walker generates tables based on specified covariates.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java [hunk 5]
```diff
@@ -33,16 +33,68 @@
...
+ * probability of error given the particular covariates seen at this site, where p(error) = num mismatches / num observations.
+ * The output file is a table (of the several covariate values, num observations, num mismatches, empirical quality score).
+ * <p>
+ * Note: ReadGroupCovariate and QualityScoreCovariate are required covariates and will be added for the user regardless of whether or not they were specified.
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: I believe this comment is outdated.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java [hunk 6]
```diff
@@ -33,16 +33,68 @@
...
+ *     <li>The recalibration table by quality score</li>
+ *     <li>The recalibration table for all the optional covariates</li>
+ * </ol>
+ *
+ * The GATK Report is intended to be easy to read by humans or computers. Check out the documentation of the GATKReport to learn how to manipulate this table.
+ * </p>
+ *
+ * <h3>Examples</h3>
+ * <pre>
+ * ./gatk BaseRecalibratorSpark \
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: `./gatk` --> `gatk`

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java [hunk 7]
```diff
@@ -33,16 +33,68 @@
...
+ *   -I my_reads.bam \
+ *   -R resources/Homo_sapiens_assembly18.fasta \
+ *   --known-sites bundle/hg18/dbsnp_132.hg18.vcf \
+ *   --known-sites another/optional/setOfSitesToMask.vcf \
+ *   -o recal_data.table
+ * </pre>
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Again, I don't think this command will run as is. Have you tested these?

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSparkSharded.java [hunk 8]
```diff
@@ -43,16 +43,58 @@
...
+ * <li>The recalibration table by read group</li>
+ * <li>The recalibration table by quality score</li>
+ * <li>The recalibration table for all the optional covariates</li>
+ * </ol>
+ * <p>
+ * The GATK Report is intended to be easy to read by humans or computers. Check out the documentation of the GATKReport to learn how to manipulate this table.
+ * </p>
+ * <p>
+ * <h3>Examples</h3>
+ * <pre>
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSparkSharded.java]: Please update the program group to ReadDataProgramGroup

Examples --> Usage example
`./gatk` --> `gatk`

Please test your example commands to ensure they work. 

In the `All hands on deck` issue ticket, I tell Ted that if his commands work without explicitly specifying Spark options, then he should add the following blurb:

```
This tool can be run without explicitly specifying Spark options. That is to say, the given example command without Spark options will run locally. See <a href ="https://software.broadinstitute.org/gatk/documentation/article?id=10060">Tutorial#10060</a> for an example of how to set up and run a Spark tool on a cloud Spark cluster.
```

File: src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/BQSRPipelineSpark.java [hunk 9]
```diff
@@ -33,33 +33,58 @@
 import java.util.List;
 
 /**
- * BQSR. The final result is analysis-ready reads.
+ * The full BQSR pipeline in one tool to run on Spark.
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/BQSRPipelineSpark.java]: Update program group and fix the example command

File: src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java [hunk 10]
```diff
@@ -70,7 +70,7 @@
     @Argument(doc = "whether to perform alignment using BWA-MEM", shortName = "align", fullName = "align", optional = true)
     private boolean align;
 
-    @Argument(doc = "the known variants", shortName = "knownSites", fullName = "knownSites", optional = false)
+    @Argument(doc = "the known variants", shortName = "known", fullName = "known-sites", optional = false)
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java]: Update program group and add an example command please.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSR.java [hunk 11]
```diff
@@ -45,10 +45,10 @@
  *
  * <h3>Usage example</h3>
  * <pre>
- * ./gatk-launch ApplyBQSR \
+ * ./gatk ApplyBQSR \
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSR.java]: Update program group, see my comments to ApplyBQSRSpark and update the example command. For example, notice how I removed the reference from the ApplyBQSRSpark command. Example commands should represent the most straightforward way to run the tool and (hopefully) encompass the most typical use-cases.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibrator.java [hunk 12]
```diff
@@ -66,23 +66,27 @@
  *
  * <h3>Examples</h3>
  * <pre>
- * java -Xmx4g -jar GenomeAnalysisTK.jar \
- *   -T BaseRecalibrator \
+ * ./gatk BaseRecalibrator \
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibrator.java]: Update program group, see my comments to ApplyBQSRSpark and update the example command.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/Concordance.java [hunk 13]
```diff
@@ -46,7 +46,7 @@
  * <h3>Usage examples</h3>
  *
  * <pre>
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/Concordance.java]: The program group should be VariantEvaluationProgramGroup. Please streamline the example command.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/CountFalsePositives.java [hunk 14]
```diff
@@ -20,8 +20,8 @@
 import java.util.List;
 
 @CommandLineProgramProperties(
-        summary = "Count PASS (false positive) variants in a vcf file for Mutect2 NA12878 normal-normal evaluation",
-        oneLineSummary = "Count PASS variants",
+        summary = CountFalsePositives.USAGE_SUMMARY,
+        oneLineSummary = CountFalsePositives.USAGE_ONE_LINE_SUMMARY,
         programGroup = VariantProgramGroup.class
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/CountFalsePositives.java]: The program group should be VariantEvaluationProgramGroup. The doc portion including the example command isn't showing up in the javaDoc. Rather it looks like:

screenshot 2018-01-03 21 57 42

**Comment by sooheelee** [comment]
Back to you @lucidtronix.

**Comment by droazen** [comment]
Do not merge -- I'm taking this & the other argument-renaming branches over to modify them for consistency and get them merged before Monday.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java [hunk 15]
```diff
@@ -18,13 +18,18 @@
 import org.broadinstitute.hellbender.utils.read.GATKRead;
 import org.broadinstitute.hellbender.utils.recalibration.RecalibrationReport;
 
-@CommandLineProgramProperties(summary="Apply Base Quality Recalibration to a SAM/BAM/CRAM file using Spark",
-        oneLineSummary="ApplyBQSR on Spark",
-        programGroup = SparkProgramGroup.class)
+@CommandLineProgramProperties(
```
[the remark this answers is not in the document]
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java [hunk 16]
```diff
@@ -18,13 +18,18 @@
...
+        summary=ApplyBQSRSpark.USAGE_SUMMARY,
+        oneLineSummary=ApplyBQSRSpark.USAGE_ONE_LINE_SUMMARY,
+        programGroup = SparkProgramGroup.class
```
[the remark this answers is not in the document]
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java [hunk 17]
```diff
@@ -18,13 +18,18 @@
...
+)
+
 @DocumentedFeature
 @BetaFeature
 public final class ApplyBQSRSpark extends GATKSparkTool {
     private static final long serialVersionUID = 0l;
+    static final String USAGE_ONE_LINE_SUMMARY = "Apply base quality score recalibration on Spark.";
```
[the remark this answers is not in the document]
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/ApplyBQSRSpark.java]: Fixed.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 3 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Fixed.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 2 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Fixed.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 5 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Removed

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 4 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Fixed.

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 6 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 7 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Added spark arguments

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 1 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSparkSharded.java (hunk 8 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSparkSharded.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/BQSRPipelineSpark.java (hunk 9 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/BQSRPipelineSpark.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java (hunk 10 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSR.java (hunk 11 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSR.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibrator.java (hunk 12 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibrator.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/Concordance.java (hunk 13 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/Concordance.java]: Fixed

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/CountFalsePositives.java (hunk 14 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/CountFalsePositives.java]: Fixed

**Comment by droazen** [comment]
@lucidtronix When you update your branch with changes from master, use `git fetch && git rebase origin/master`, and not `git merge` or `git pull`. The reason is that the latter commands create merge commits in your history, and any conflict resolution done in those merge commits is lost during the final rebase of the branch.

Also, please don't merge this into master yet -- I've been making some changes locally to all the arg renaming PRs for consistency reasons.

**Comment by lucidtronix** [comment]
@droazen Sorry! just saw your comments now... I was trying git's resolve conflicts button but apparently it uses git merge. This branch is all yours of course!

File: src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java (hunk 7 above)
Comment (sooheelee) [reply src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java]: Thanks!

File: src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java [hunk 18]
```diff
@@ -44,22 +44,50 @@
...
+ * </pre>
+ * <p>
+ * To additionally align reads with BWA-MEM:
+ * </p>
+ * <pre>
+ * gatk ReadsPipelineSpark \
+ *   -I gs://my-gcs-bucket/unaligned_reads.bam \
+ *   -R gs://my-gcs-bucket/reference.fasta \
+ *   --known-sites gs://my-gcs-bucket/sites_of_variation.vcf \
+ *   -align
```
Comment (sooheelee) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java]: needs a backslash \

**Comment by sooheelee** [comment]
Thanks for incorporating feedback. Everything looks good except I think since I last looked at this PR, `--sparkRunner` has become `--spark-runner`. If this will get fixed, then there is a tiny thing to fix in addition. The ReadsPipelineSpark example command is missing a backslash.

**Comment by droazen** [comment]
I'm going to rebase and patch this branch now -- it conflicts with another arg-renaming branch that I just merged.

**Pull request merged**